tests/isolation_provider/base.py

```diff
@@ -164,6 +164,7 @@ def test_linger_unkillable(
         terminate_proc_mock = mocker.patch.object(
             provider, "terminate_doc_to_pixels_proc", return_value=None
         )
+        kill_pg_orig = base.kill_process_group
```

Comment:
I'm surprised: I thought that the whole reason for using pytest mock was so we don't have to do this ourselves.

Reply:
Ah: I get it with your initial PR description (maybe commit message): you need it once more after the original patch, but not after the test ends.

That looks good to me then. Note that we could also use it as a context manager (`with`) if we wanted.